dimod/decorators.py

```diff
@@ -497,3 +498,21 @@ def wrapper(obj, *args, **kwargs):
         return method(*args, **kwargs)
 
     return wrapper
+
+def unique_variable_labels(f):
+    """Decorator to assign unique labels to variables when no label is passed.
+
+    Designed to be applied to variable methods, :meth:`.dimod.Binary`,
+    :meth:`.dimod.Spin`, :meth:`.dimod.Integer`.
+
+    """
+    @wraps(f)
+    def conditional_unique_label(*args, **kwargs):
+        if len(args) == 0 and len(kwargs) == 0:
```

Comment:
why do we want `len(kwargs) == 0`? It seems that we don't because I believe `Integer(lower_bound=1)` will fail as currently implemented.

Reply:
That is right..Was definitely looking at the signatures of `Spin` and `Binary` when writing this but it would break with those keyword only args on `Integer`. Have to rethink along with the above comment.

Reply:
Reworked this. Please double check the new logic!